src/Framework/PaymentGateways/DonationSummary.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Give\Framework\PaymentGateways;
+
+use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
+
+/**
+ * @unreleased
+ */
+class DonationSummary
+{
+    /** @var int */
+    protected $length = 255;
+
+    /** @var GatewayPaymentData */
+    protected $paymentData;
+
+    /**
+     * @unreleased
+     * @param GatewayPaymentData $paymentData
+     */
+    public function __construct( GatewayPaymentData $paymentData )
+    {
+        $this->paymentData = $paymentData;
+    }
+
+    /**
+     * @unreleased
+     * @param $length
+     */
+    public function setLength( $length )
+    {
+        $this->length = $length;
+    }
+
+    /**
+     * @unreleased
+     * @return string
+     */
+    public function getSummaryWithDonor()
+    {
+        return $this->trimAndFilter(implode(' - ', [
+            $this->getSummary(),
+            $this->getDonorLabel(),
+        ]));
+    }
+
+    /**
+     * @unreleased
+     * @return string
+     */
+    public function getSummary()
+    {
+        return $this->trimAndFilter(implode( ': ', array_filter([
+            $this->getLabel(),
+            $this->getPriceLabel(),
+        ])));
+    }
+
+    /**
+     * @unreleased
+     * @param $property
+     * @return mixed|void
+     */
+    protected function get( $property )
+    {
+        if( property_exists( $this->paymentData, $property ) ) {
+            return $this->paymentData->$property;
+        }
+    }
+
+    /**
+     * @unreleased
+     * @return string
+     */
+    protected function getLabel()
+    {
+        $formId = give_get_payment_form_id( $this->get( 'donationId' ) );
+        return wp_sprintf( __( 'Donation Form ID: %d', 'give' ), $formId );
```

Comment:
@kjohnson any specific reason to use `wp_sprintf`?

Reply:
Interesting: https://andy.wordpress.com/2011/01/05/wp_sprintf/

Reply:
@kjohnson interesting read.

I think we can replace `wp_sprintf ` with `sprintf`.

Reply:
@ravinderk updated, see https://github.com/impress-org/givewp/pull/6209/commits/934de94a40a511032b38dc8049170dbb4eea57b1